core/quantity/src/Parser.ts

```diff
@@ -712,29 +723,38 @@ export class Parser {
     let matchedSuffix: DirectionLabel | null = null;
 
     // check if input string begins with northLabel or southLabel and strip it off
-    if (inString.startsWith(DirectionLabel.North)) {
+    if (inString.toUpperCase().startsWith(DirectionLabel.North)) {
       inString = inString.substring(DirectionLabel.North.length);
       matchedPrefix = DirectionLabel.North;
-    } else if (inString.startsWith(DirectionLabel.South)) {
+    } else if (inString.toUpperCase().startsWith(DirectionLabel.South)) {
       inString = inString.substring(DirectionLabel.South.length);
       matchedPrefix = DirectionLabel.South;
     }
 
-    // check if input string ends with eastLabel or westLabel and strip it off
-    if (inString.endsWith(DirectionLabel.East)) {
+    // check if input string ends with eastLabel or westLabel (case-insensitive) and strip it off
+    if (inString.toUpperCase().endsWith(DirectionLabel.East)) {
       inString = inString.substring(0, inString.length - DirectionLabel.East.length);
       matchedSuffix = DirectionLabel.East;
-    } else if (inString.endsWith(DirectionLabel.West)) {
+    } else if (inString.toUpperCase().endsWith(DirectionLabel.West)) {
       inString = inString.substring(0, inString.length - DirectionLabel.West.length);
       matchedSuffix = DirectionLabel.West;
     }
 
+    // check if the remaining string is a special direction
+    if (inString.trim() === "") {
+      const prefix = matchedPrefix?.toLowerCase() || "";
+      const suffix = matchedSuffix?.toLowerCase() || "";
+      const specialDirection = specialDirections[`${prefix}${suffix}`];
+      if (specialDirection !== undefined)
+        return { ok: true, value: specialDirection };
+    }
+
     if (matchedPrefix === null || matchedSuffix === null) {
       return { ok: false, error: ParseError.BearingPrefixOrSuffixMissing };
     }
 
     const parsedResult = this.parseAndProcessTokens(inString, spec.format, spec.unitConversions);
-    if(this.isParseError(parsedResult) || !parsedResult.ok) {
+    if(this.isParseError(parsedResult)) {
```

Comment:
maybe also remove the RHS of the `||` at the start of `parseAzimuthFormat()` as well, to be consistent